apps/internal-ui/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,50 @@
+import clsx from 'clsx';
+import { useId } from 'react';
+
+import CheckboxIcon from '@/components/Checkbox/CheckboxIcon';
+import CheckboxLabel from '@/components/Checkbox/CheckboxLabel';
+import { CHECKBOX_SIZES } from '@/components/Checkbox/constants';
+import { CheckboxProps } from '@/components/Checkbox/types';
+
+const Checkbox = ({
+  checked,
+  onChange,
+  disabled,
+  label,
+  size = CHECKBOX_SIZES.SM,
+  className,
+}: CheckboxProps) => {
+  const id = useId();
+  const labelId = `${id}-label`;
+
+  return (
+    <label
+      className={clsx(
+        className,
+        'group',
+        !disabled ? 'cursor-pointer' : 'cursor-not-allowed',
+      )}
+      htmlFor={id}
+    >
+      <input
+        aria-labelledby={labelId}
+        checked={checked}
+        className='hidden'
+        disabled={disabled}
+        id={id}
+        type='checkbox'
+        onChange={onChange}
+      />
+      {label ? (
+        <div className='flex items-center gap-x-2'>
+          <CheckboxIcon size={size} />
+          <CheckboxLabel label={label} labelId={labelId} size={size} />
+        </div>
+      ) : (
+        <CheckboxIcon size={size} />
+      )}
```

Comment:
```suggestion
      <CheckboxIcon size={size} />
      {label && <CheckboxLabel label={label} labelId={labelId} size={size} />}
```

q: label 에 flex 주지 않고 div 를 추가해서 감싸신 이유가 따로 있는거죠??

Reply:
불필요한 부모 컨테이너를 사용하지 않기 위해서입니다!